canonicalwebteam/flask_base/app.py

```diff
@@ -17,32 +21,71 @@ def __init__(
         self,
         name,
         service,
-        prepare_deleted=prepare_deleted(),
-        prepare_redirects=prepare_redirects(),
+        favicon_url=None,
+        template_404=None,
+        template_500=None,
         *args,
         **kwargs
     ):
         super().__init__(name, *args, **kwargs)
 
         self.service = service
 
-        self.config.from_object(
-            "canonicalwebteam.flask_base.config.BaseConfig"
-        )
+        self.config["SECRET_KEY"] = os.getenv("SECRET_KEY", "base_secret")
 
         self.url_map.strict_slashes = False
         self.url_map.converters["regex"] = RegexConverter
 
-        self.wsgi_app = (
-            ProxyFix(self.wsgi_app)
-            if not self.debug
-            else DebuggedApplication(self.wsgi_app)
-        )
+        self.wsgi_app = ProxyFix(self.wsgi_app)
 
-        self.before_request(prepare_redirects)
-        self.before_request(prepare_deleted)
+        if self.debug:
+            self.wsgi_app = DebuggedApplication(self.wsgi_app)
+
+        self.before_request(prepare_redirects())
+        self.before_request(
+            prepare_redirects(
+                path=os.path.join(self.root_path, "redirects.yaml"),
+                permanent=True
+            )
+        )
+        self.before_request(prepare_deleted())
 
         self.context_processor(base_context)
 
         talisker.flask.register(self)
         talisker.logs.set_global_extra({"service": self.service})
+
+        # Default error handlers
+        if template_404:
+
+            @self.errorhandler(404)
+            def not_found_error(error):
+                return flask.render_template(template_404), 404
+
+        if template_500:
+
+            @self.errorhandler(500)
+            def internal_error(error):
+                return flask.render_template(template_500), 500
+
+        # Default routes
+        if favicon_url:
+
+            @self.route("/favicon.ico")
+            def favicon():
+                return flask.redirect(favicon_url)
+
+        robots_path = os.path.join(self.root_path, 'robots.txt')
+        humans_path = os.path.join(self.root_path, 'humans.txt')
+
+        if os.path.isfile(robots_path):
+
+            @self.route("/robots.txt")
+            def robots():
+                flask.send_file(robots_path)
```

Comment:
I believe this needs to be `return flask.send_file(robots_path)`.

Reply:
Yep, I shouldn't have pushed so early =)